test/unit/ngsiv2/plugins/bidirectional-plugin_test.js

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright 2016 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-iotagent-lib
+ *
+ * fiware-iotagent-lib is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-iotagent-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::daniel.moranjimenez@telefonica.com
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+
+'use strict';
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    logger = require('logops'),
+    nock = require('nock'),
+    request = require('request'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {
```

Comment:
As far as I understand, provisionBidirectionalDevice.json uses TheLightType. Thus, is this really needed this `types` here?

Reply:
Fixed in 7a344dea98034de3e0144094864785f14d44bf64